File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    max_w = 0
    max_h = 0

    for w, h in sizes:
        big_side = max(w, h)
        small_side = min(w, h)
        max_w = max(max_w, big_side)
        max_h = max(max_h, small_side)

    return max_w * max_h
    
    
    

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
const solution = (sizes) => {
    let width = 0;
    let height = 0;
    
    sizes.forEach((lst) => {
        let large_side = Math.max(...lst);
        let small_side = Math.min(...lst);
        width = Math.max(width, large_side);
        height = Math.max(height, small_side);
    });
    return width * height;
}